packages/compiler/src/server/lib-provider.ts

```diff
@@ -1,56 +1,70 @@
 import { joinPaths } from "../core/path-utils.js";
 import { NpmPackage, NpmPackageProvider } from "./npm-package-provider.js";
 
-export class EmitterProvider {
+export class LibraryProvider {
   private isEmitterPackageCache = new Map<string, boolean>();
+  private isLinterPackageCache = new Map<string, boolean>();
```

Comment:
Please make the Libraryprovider a general provider which doesnt have concept of Emitter or Linter. Then you can create instances of it for emitter and linter.